src/lib/caching/structures/messages/Message.ts

```diff
@@ -278,6 +279,22 @@ export class Message extends WebhookMessage<Client> {
 		return this.content;
 	}
 
+	/**
+	 * Pins the message to the channel
+	 * @since 0.0.4
+	 */
+	public async pin(): Promise<Message> {
+		return this.channel.pins.add(this.id);
+	}
+
+	/**
+	 * Unpins the message to the channel
+	 * @since 0.0.4
+	 */
+	public async unpin(): Promise<ChannelPinsStore> {
+		return this.channel.pins.remove(this.id);
```

Comment:
Do we want to check if this.pinned at other devs?

Reply:
Well, if we start doing certain checks for the end user, we might as well throw in permission checks in the lib before sending requests to Discord.. I say it's fine